autosklearn/pipeline/components/regression/mlp.py

```diff
@@ -0,0 +1,239 @@
+import numpy as np
+
+from ConfigSpace.configuration_space import ConfigurationSpace
+from ConfigSpace.hyperparameters import UniformFloatHyperparameter, \
+    UniformIntegerHyperparameter, UnParametrizedHyperparameter, Constant, \
+    CategoricalHyperparameter
+from ConfigSpace.conditions import InCondition
+
+from autosklearn.pipeline.components.base import (
+    AutoSklearnRegressionAlgorithm,
+    IterativeComponent,
+)
+from autosklearn.pipeline.constants import SPARSE, DENSE, UNSIGNED_DATA, PREDICTIONS
+from autosklearn.util.common import check_for_bool
+
+
+class MLPRegressor(
+    IterativeComponent,
+    AutoSklearnRegressionAlgorithm
+):
+    def __init__(self, hidden_layer_depth, num_nodes_per_layer, activation, alpha,
+                 learning_rate_init, early_stopping, solver, batch_size,
+                 n_iter_no_change, tol, shuffle, beta_1, beta_2, epsilon,
+                 validation_fraction=None, random_state=None, verbose=0):
+        self.hidden_layer_depth = hidden_layer_depth
+        self.num_nodes_per_layer = num_nodes_per_layer
+        self.max_iter = self.get_max_iter()
+        self.activation = activation
+        self.alpha = alpha
+        self.learning_rate_init = learning_rate_init
+        self.early_stopping = early_stopping
+        self.n_iter_no_change = n_iter_no_change
+        self.validation_fraction = validation_fraction
+        self.tol = tol
+        self.solver = solver
+        self.batch_size = batch_size
+        self.shuffle = shuffle
+        self.beta_1 = beta_1
+        self.beta_2 = beta_2
+        self.epsilon = epsilon
+        self.beta_1 = beta_1
+
+        self.random_state = random_state
+        self.verbose = verbose
+        self.estimator = None
+
+    @staticmethod
+    def get_max_iter():
+        return 512
+
+    def get_current_iter(self):
+        return self.estimator.n_iter_
+
+    def iterative_fit(self, X, y, n_iter=2, refit=False):
+        """
+        Set n_iter=2 for the same reason as for SGD
+        """
+        from sklearn.neural_network import MLPRegressor
+        import sklearn.preprocessing
+        n_iter = max(n_iter, 2)
+
+        if refit:
+            self.estimator = None
+            self.scaler = None
+
+        if self.estimator is None:
+            self._fully_fit = False
+
+            self.max_iter = int(self.max_iter)
+            self.hidden_layer_depth = int(self.hidden_layer_depth)
+            self.num_nodes_per_layer = int(self.num_nodes_per_layer)
+            self.hidden_layer_sizes = tuple(self.num_nodes_per_layer
+                                            for i in range(self.hidden_layer_depth))
+            self.activation = str(self.activation)
+            self.alpha = float(self.alpha)
+            self.learning_rate_init = float(self.learning_rate_init)
+            self.early_stopping = str(self.early_stopping)
```

Comment:
This 1st cast isn't there for classification -> I'm not sure which one is correct.